src/json-schema/partial/map_state.json

```diff
@@ -49,6 +49,219 @@
+                                            "description": "A reference path to an integer that limits the number of data items passed to the Map state."
+                                        }
+                                    },
+                                    "required": [
+                                        "InputType"
+                                    ]
+                                },
+                                "Resource": {
+                                    "type": "string",
+                                    "description": "The Amazon S3 API action Step Functions must invoke depending on the specified dataset.",
+                                    "examples": [
+                                        "arn:aws:states:::s3:getObject"
+                                    ]
+                                },
+                                "Parameters": {
+                                    "type": "object",
+                                    "description": "A JSON object that specifies the Amazon S3 bucket name and object key in which the dataset is stored.",
+                                    "properties": {
+                                        "Bucket": {
+                                            "type": "string",
+                                            "description": "A string that specifies the Amazon S3 bucket name in which the dataset is stored.",
+                                            "minLength": 1
+                                        },
+                                        "Key": {
+                                            "type": "string",
+                                            "description": "A string that specifies Amazon S3 object key in which the dataset is stored.",
+                                            "minLength": 1
+                                        }
+                                    },
+                                    "required": [
+                                        "Bucket", 
+                                        "Key"
+                                    ]
+                                }
+                            },
+                            "required": [
+                                "Resource",
+                                "Parameters"
+                            ]
+                        },
+                        "ItemSelector": {
+                            "type": "object",
+                            "description": "A JSON object that overrides each single element of the item array."
+                        },
+                        "ItemBatcher": {
+                            "type": "object",
+                            "description": "A JSON object that specifies how to batch the items for the ItemProcessor.",
+                            "properties": {
+                                "BatchInput": {
```

Comment:
Do we have subtypes for this property?

Reply:
No, it just has to be a valid JSON object.

https://docs.aws.amazon.com/step-functions/latest/dg/input-output-itembatcher.html